### ISSUE TITLE
Squiz.Commenting.FunctionComment.IncorrectTypeHint recognizes generics a part of type hints

### ISSUE THREAD

**Issue by Karpec** [PHPCSStandards/PHP_CodeSniffer#746]
## Describe the bug
Hi, I ran into this issue again. which was already [ addressed](https://github.com/squizlabs/PHP_CodeSniffer/issues/3589) in the old package. Even PR is mentioned as a likely solution. Can we get it into the production version if possible? Thank you for your time :)

### Code sample
```php
use Illuminate\Support\Collection;

/**
 * @param array<int, string>                          $array
 * @param iterable<string>                            $iterable
 * @param int<0, 100>                                 $int
 * @param \Illuminate\Support\Collection<int, string> $collection
 *
 * @return \Illuminate\Support\Collection<int, string>
 */
function test(
    array $array,
    iterable $iterable,
    int $int,
    Collection $collection,
): Collection {
    return new Collection();
}
```

### Custom ruleset
N/A

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```text
 15 | ERROR   | [ ] Expected type hint "iterable<string>"; found "iterable" for $iterable (Squiz.Commenting.FunctionComment.IncorrectTypeHint)
 15 | ERROR   | [ ] Expected type hint "int<0, 100>"; found "int" for $int (Squiz.Commenting.FunctionComment.IncorrectTypeHint)
 15 | ERROR   | [ ] Expected type hint "\Illuminate\Support\Collection<int, string>"; found "Collection" for $collection
    |         |     (Squiz.Commenting.FunctionComment.IncorrectTypeHint)
```

## Expected behavior

Errors should not occur with generics <>, just like how it works for an array

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | [MacOS 15.1.1 - docker container]                                                       |
| PHP version                    | [8.1]                                                                                     |
| PHP_CodeSniffer version | [3.11.1, master]                                                                            |
| Standard                         | [Squiz]                                                      |
| Install type                      | [Composer] |

**Comment by asispts** [comment]
I don't think `phpcs` is the right tool for this case. This should be handled by a static analyzer.

IMHO, this rule could be deprecated/removed.

**Comment by Karpec** [comment]
Yeah, that's when I stumbled upon it, when leveling up PHPstan to 6. I might turn this rule off eventually, although sometimes it's better to double check

**Comment by jrfnl** [comment]
@Karpec I'd be happy to have a look at a PR. Maybe @tm1000 would be willing to port their original PR over to this repo (and add some extra tests based on the code sample above) ?

@asispts Well, PHPCS _is_ a static analysis tool ;-)

**Comment by Karpec** [comment]
@jrfnl I'll create a PR for it today, I have time :)

**Comment by Karpec** [comment]
PR created :)

**Comment by rodrigoprimo** [comment]
I stumbled upon this issue while triaging some issues in the old repository. Just leaving here two things that might need to be checked when working on a fix for this issue that I don't think were considered in #750.

Generics can be nested, and the regex on [the closed PR](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/6313cbf64/src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php#L437) doesn't seem to take that into account. It uses `[^>]+` to match the generic part, which fails for nested generics like `Collection<int, Map<string, mixed>>` because `[^>]+` stops at the first `>`.

```php
@param Collection<int, Map<string, mixed>>    $nestedGeneric A nested generic param.
```

Besides that, while testing, I noticed that the `strpos($suggestedName, 'array')` check on [line 431](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/6313cbf64/src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php#L431) runs before the regex and matches any type containing "array" anywhere in the string. This means a type like `Collection<int, array<string, mixed>>` incorrectly gets `$suggestedTypeHint = 'array'` instead of `Collection`, producing a false positive `IncorrectTypeHint` error.

```php
@param Collection<int, array<string, mixed>>  $nestedArray   A nested generic param.
```
